### ISSUE TITLE
MultiNodeTreePicker2 not working in DocTypeGridEditorSurfaceController

### ISSUE THREAD

**Issue by leekelleher** [umco/umbraco-doc-type-grid-editor#124]
Copied bug report from Our Umbraco forum by @herpoelaertw. Opening on here for further investigation...

> When I try to get the value of a Content picker (MultiNodeTreePicker2) in DocTypeGridEditorSurfaceController, I receive the following error:
>
> I'm using the latest version of Umbraco (7.12.4) and DTGE (0.6.0). With version 0.5.0 everything was working fine.

https://our.umbraco.com/packages/backoffice-extensions/doc-type-grid-editor/doc-type-grid-editor-feedback/95168-content-picker-not-working-in-doctypegrideditorsurfacecontroller

<details>
    <summary>Stack Trace</summary>

```
Value cannot be null. Parameter name: g
    at System.Guid..ctor(String g)
    at umbraco.presentation.preview.PreviewContent.EnsureInitialized(User user, String previewSet, Boolean validate, Action initialize)
    at Umbraco.Web.PublishedCache.XmlPublishedCache.PublishedContentCache.<>c.<get_GetXmlDelegate>b__42_0(UmbracoContext context, Boolean preview)
    at Umbraco.Web.PublishedCache.XmlPublishedCache.PublishedContentCache.GetXml(UmbracoContext umbracoContext, Boolean preview)
    at Umbraco.Web.PublishedCache.XmlPublishedCache.PublishedContentCache.GetById(UmbracoContext umbracoContext, Boolean preview, Int32 nodeId)
    at Umbraco.Web.PublishedCache.XmlPublishedCache.PublishedContentCache.GetById(UmbracoContext umbracoContext, Boolean preview, Guid nodeKey)
    at Umbraco.Web.PublishedCache.ContextualPublishedContentCache.GetById(Boolean preview, Guid contentKey)
    at Umbraco.Web.PublishedContentQuery.TypedContent(Guid id)
    at Umbraco.Web.PublishedContentQueryExtensions.TypedContent(ITypedPublishedContentQuery contentQuery, Udi id)
    at Umbraco.Web.PropertyEditors.ValueConverters.MultiNodeTreePickerPropertyConverter.GetPublishedContent[T](T nodeId, UmbracoObjectTypes& actualType, UmbracoObjectTypes expectedType, Func`2 contentFetcher)
    at Umbraco.Web.PropertyEditors.ValueConverters.MultiNodeTreePickerPropertyConverter.ConvertSourceToObject(PublishedPropertyType propertyType, Object source, Boolean preview)
    at Our.Umbraco.DocTypeGridEditor.Models.DetachedPublishedProperty.<.ctor>b__7_1()
    at System.Lazy`1.CreateValue()
    at System.Lazy`1.LazyInitValue()
    at Umbraco.Web.PublishedPropertyExtension.GetValue[T](IPublishedProperty property, Boolean withDefaultValue, T defaultValue)
    at Umbraco.Web.PublishedContentExtensions.GetPropertyValue[T](IPublishedContent content, String alias)
    at WebApplication4.App_Code.TestSurfaceController.Test() in c:\Users\WP-27\documents\visual studio 2017\projects\webapplication4\webapplication4\App_Code\TestSurfaceController.cs:line 16
    at lambda_method(Closure , ControllerBase , Object[] )
    at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
    at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
    at System.Web.Mvc.Async.AsyncControllerActionInvoker.<BeginInvokeSynchronousActionMethod>b__39(IAsyncResult asyncResult, ActionInvocation innerInvokeState)
    at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`2.CallEndDelegate(IAsyncResult asyncResult)
    at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
    at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3d()
    at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f()
```
</details>

**Comment by leekelleher** [comment]
I've investigated this one, and it's a difficult one to fix.

Unknowingly, by introducing the `InPreviewMode = true` (in commit 0ebaf166c4233274b6f43d6897ede5467be9f8b9 for v0.6.0), this has increased the complexity of the preview mechanism.  Any calls to the content cache API will check the `InPreviewMode` value and attempt to populate the XML content cache from the preview (unpublished node).

The difficulty for DTGE comes with that each individual grid cell that has a preview is only aware of itself, meaning that if we attempt to create the preview XML, we'd be doing so for each and every DTGE grid cell. The performance (and processing) of this alone would be horrific.

I suspect that we'll need to revert setting `InPreviewMode`, which would break #117 (for Vorto), and figure out another way of fixing that.

**Comment by leekelleher** [comment]
After a lot of digging around in the Umbraco core source code, I found that manually setting `InPreviewMode` causes more headaches than it's worth. So I've reverted a things and hopefully fixed it in PR #127.

Once PR #127 is merged it, it should fix this specific bug.  I'll close off this ticket for now.

**Comment by rich-hamilton** [comment]
Thanks Lee,
Was pulling my hair out on Friday with this problem. just upgraded this morning and all good.
:)
